paasta_tools/setup_istio_mesh.py

```diff
@@ -0,0 +1,235 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Usage: ./setup_kubernetes_services.py <service.instance>
+
+Command line options:
+
+- -c, --cluster: PAASTA cluster
+- -v, --verbose: Verbose output
+"""
+import argparse
+import json
+import logging
+import sys
+from typing import Mapping
+from typing import Sequence
+
+import kubernetes.client as k8s
+
+from paasta_tools.kubernetes_tools import ensure_namespace
+from paasta_tools.kubernetes_tools import KubeClient
+from paasta_tools.kubernetes_tools import KubernetesService
+from paasta_tools.kubernetes_tools import sanitise_kubernetes_name
+from paasta_tools.utils import SPACER
+
+log = logging.getLogger(__name__)
+
+UNIFIED_K8S_SVC_NAME = "paasta-routing"
+UNIFIED_SVC_PORT = 1337
+PAASTA_SVC_PORT = 8888
+PAATA_REGISTRATION_PREFIX = "registrations.paasta.yelp.com"
+PAASTA_SERVICE_FILE = "/nail/etc/"
+PAASTA_NAMESPACE = "paasta"
+
+
+class ErrorCreatingUnifiedService(Exception):
+    pass
+
+
+class ErrorGettingServiceList(Exception):
+    pass
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="Creates Kubernetes services.")
+    parser.add_argument(
+        "service_instance_list",
+        nargs="+",
+        help="The list of Kubernetes service instances to create or update services for",
+        metavar="SERVICE%sINSTANCE" % SPACER,
+    )
+    parser.add_argument(
+        "-v", "--verbose", action="store_true", dest="verbose", default=False,
+    )
+
+    parser.add_argument(
+        "--dry-run", action="store_true", dest="dry_run", default=False,
+    )
+
+    parser.add_argument(
+        "-l",
+        "--rate-limit",
+        dest="rate_limit",
+        default=0,
+        metavar="LIMIT",
+        type=int,
+        help="Update or create up to this number of service instances. Default is 0 (no limit).",
+    )
+    parser.add_argument(
+        "-f",
+        "--file-path",
+        dest="file_path",
+        default="/nail/etc/services/services.json",
+        metavar="LIMIT",
+        type=str,
+        help="services file to use for smartstack namespace. Default file is /nail/etc/services/services.json",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def build_namespace_port_mapping(file_path: str) -> Mapping:
+    paasta_namespaces = {}
+    with open(file_path) as service_file:
+        smartstack_namespaces = json.load(service_file)
+        for namespace, endpoint in smartstack_namespaces.items():
+            paasta_namespaces[namespace] = endpoint["port"]
```

Comment:
i don't think we need to bother with this, just return smartstack_namespaces directly, we'll know how to use them

Reply:
The reason to include the ports is that they will be needed to be set on the paasta-routing service, With this mapping we keeping the collection of ports that need to be exposed on paasta routing service

Reply:
sure, but `smartstack_namespaces` already contains ports so we don't need to rebuild the same hashmap